static/js/browser2/out_arcs.tsx

```diff
@@ -0,0 +1,157 @@
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from "react";
+import axios from "axios";
+import _ from "lodash";
+
+const ignoredOutArcProperties = new Set([
+  "provenance",
+  "kmlCoordinates",
+  "geoJsonCoordinates",
+  "geoJsonCoordinatesDP1",
+  "geoJsonCoordinatesDP2",
+  "geoJsonCoordinatesDP3",
+]);
+const maxNameWithLanguageValues = 10;
+
+interface OutArcsSectionPropType {
+  dcid: string;
+  outArcLabels: Array<string>;
+  provDomain: { [key: string]: URL };
+}
+
+interface OutArcsSectionStateType {
+  //TODO (chejennifer): replace the any type with an actual type
+  arcValuesData: { [predicate: string]: Array<any> };
+  propertyLabels: Array<string>;
+}
+
+export class OutArcsSection extends React.Component<
+  OutArcsSectionPropType,
+  OutArcsSectionStateType
+> {
+  constructor(props: OutArcsSectionPropType) {
+    super(props);
+    this.state = {
+      arcValuesData: {},
+      propertyLabels: [],
+    };
+  }
+
+  componentDidMount(): void {
+    this.fetchData();
+  }
+
+  private fetchData(): void {
+    const propValuesPromises = this.props.outArcLabels.map((label) => {
+      if (!ignoredOutArcProperties.has(label)) {
+        return axios
+          .get(`/api/browser/propvals/${label}/${this.props.dcid}`)
+          .then((resp) => resp.data);
+      }
+    });
+    Promise.all(propValuesPromises).then((propValuesData) => {
+      const outArcsByPredicate = {};
+      const propertyLabels = [];
+      propertyLabels.push("typeOf");
+      propValuesData.forEach((valuesData) => {
+        if (!valuesData || _.isEmpty(valuesData.values)) {
+          return;
+        }
+        const predicate = valuesData.property;
+        let values = valuesData.values.out;
+        if (predicate === "nameWithLanguage") {
```

Comment:
can you make this more general, so can treat other predicates in the same way too. Maybe add a constant for special prop list

Reply:
done